netstandard/tools/HandlePackageFileConflicts.cs

```diff
@@ -155,6 +241,75 @@ private int GetPackageRank(ITaskItem item)
             return rank;
         }
 
+        /// <summary>
+        /// Get's the key to use for identifying reference conflicts
+        /// </summary>
+        /// <param name="item"></param>
+        /// <returns></returns>
+        private static string GetReferenceKey(ITaskItem item)
+        {
+            var aliases = item.GetMetadata("Aliases");
+
+            if (!String.IsNullOrEmpty(aliases))
+            {
+                // skip conflict detection for aliased assemblies
+                return null;
+            }
+
+            // We only handle references that have path information since we're only concerned
+            // with resolving conflicts between file references.  If conflicts exist between 
+            // named references that are found from AssemblySearchPaths we'll leave those to
+            // RAR to handle or not as it sees fit.
+            var sourcePath = GetSourcePath(item);
+
+            if (String.IsNullOrEmpty(sourcePath))
+            {
+                return null;
+            }
+
+            // The following is similar to Path.GetFileName but does not throw and returns null
+            // rather than empty for a directory.
+            var length = sourcePath.Length;
+            for (int i = length - 1; --i >= 0;)
+            {
+                var ch = sourcePath[i];
+
+                if (ch == Path.DirectorySeparatorChar || ch == Path.AltDirectorySeparatorChar || ch == Path.VolumeSeparatorChar)
+                {
+                    // string terminated with a separator: we were given a directory, ignore it.
+                    if (i == length - 1)
+                    {
+                        return null;
+                    }
+
+                    // we found a separator, return filename portion.
+                    var startOfFileName = i + 1;
+                    return sourcePath.Substring(startOfFileName);
+                }
+            }
+
+            // could not find a separator, return source path
+            return sourcePath;
+        }
+
+        private static string GetReferenceTargetPath(ITaskItem item)
+        {
+            // Determine if the reference will be copied local.  
+            // We're only dealing with primary file references.  For these RAR will 
+            // copy local if Private is true or unset.
+
+            var isPrivate = MSBuildUtilities.ConvertStringToBool(item.GetMetadata("Private"), defaultValue: true);
```

Comment:
That's the effective behavior for a file-based reference.  See https://github.com/Microsoft/msbuild/blob/cb8c727a898e4439a3cd18f329504e436b0b7c00/src/XMakeTasks/AssemblyDependency/Reference.cs#L1202 and https://github.com/Microsoft/msbuild/blob/cb8c727a898e4439a3cd18f329504e436b0b7c00/src/XMakeTasks/AssemblyDependency/Reference.cs#L1330